### PR TITLE
Feat/add update breakout

### PR DESCRIPTION
**Pull request by othellashao**
# COMPLETES #
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-421079

## This pull request addresses

add the missing api of updating breakout information to server

## by making the following changes

1. add update breakout api and unit test of it
2. add protection for edit lock

### Change Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [x] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/plugin-meetings/src/breakouts/index.ts [hunk 1]
```diff
@@ -581,6 +609,9 @@ const Breakouts = WebexPlugin.extend({
   keepEditLockAlive() {
     if (this.editLock && !!this.editLock.token) {
       const ttl = this.editLock.ttl < 30 ? BREAKOUTS.DEFAULT_TTL : this.editLock.ttl;
+      if (this.intervalID) {
+        window.clearInterval(this.intervalID);
```
Comment (Coread) [inline packages/@webex/plugin-meetings/src/breakouts/index.ts:613]: issue: you should add a test to make sure you clear the interval

File: packages/@webex/plugin-meetings/src/breakouts/index.ts [hunk 2]
```diff
@@ -531,7 +559,7 @@ const Breakouts = WebexPlugin.extend({
     if (breakout.body?.groups) {
       this.set('groups', breakout.body.groups);
     }
-    if (breakout.body?.editlock && editlock) {
+    if (editlock && breakout.body?.editlock?.token) {
```
Comment (Coread) [inline packages/@webex/plugin-meetings/src/breakouts/index.ts:562]: issue: I don't see a test checking this case

File: packages/@webex/plugin-meetings/src/breakouts/index.ts [hunk 3]
```diff
@@ -517,6 +517,34 @@ const Breakouts = WebexPlugin.extend({
...
+      ...{groups: [payload]},
+    };
+
+    return this.request({
+      method: HTTP_VERBS.PUT,
+      uri: this.url,
+      body,
+    }).catch((error) => {
+      return Promise.reject(
+        boServiceErrorHandler(error, 'Breakouts#update --> Edit lock token mismatch')
```
Comment (Coread) [inline packages/@webex/plugin-meetings/src/breakouts/index.ts:543]: issue: I can't see a test checking this reject

**Review by Coread — CHANGES_REQUESTED** [review]
please see comments

File: packages/@webex/plugin-meetings/src/breakouts/index.ts (hunk 3 above)
Comment (Coread) [reply packages/@webex/plugin-meetings/src/breakouts/index.ts:543]: @othellashao are you sure the message here 'Breakouts#update --> Edit lock token mismatch' is correct? Given that other errors are possible perhaps just "Breakouts#update" might be better

**Comment by othellashao** [comment]
Can't push change on it since the branch was protected. Create new PR on it

**Pull request closed without merging**